kasa/smartprotocol.py

```diff
@@ -101,7 +101,12 @@ async def _query(self, request: Union[str, Dict], retry_count: int = 3) -> Dict:
         # make mypy happy, this should never be reached..
         raise SmartDeviceException("Query reached somehow to unreachable")
 
-    async def _execute_query(self, request: Union[str, Dict], retry_count: int) -> Dict:
+    def get_method_and_params_for_request(self, request: Union[str, Dict]):
+        """Return method and parameters for the given request.
```

Comment:
Can you take a look at https://github.com/python-kasa/python-kasa/pull/662 in the context of this change?  Maybe we should merge 662 if you're happy with it and rebase this on top?

Reply:
Done :+1: